weldx/asdf/types.py

```diff
@@ -88,15 +88,18 @@ def __new__(mcs, name, bases, attrs):
 class WeldxConverter(Converter, metaclass=WeldxConverterMeta):
     """Base class to inherit from for custom converter classes."""
 
-    tags: list[str] = []
-    types: list[Union[type, str]] = []
+    tags: tuple[str] = None  # note: this will be updated by WeldxConverterMeta.
+    types: tuple[Union[type, str]] = ()
 
     def to_yaml_tree(self, obj, tag: str, ctx: SerializationContext):
         raise NotImplementedError
 
     def from_yaml_tree(self, node: dict, tag: str, ctx: SerializationContext):
         raise NotImplementedError
 
+    def select_tag(self, obj, tags, ctx):
+        return sorted(tags)[-1]
```

Comment:
Can we be sure, that the tags/objs involved are part of weldx? Should we make this assertion here, or in converters deriving from this, treating a special case?

Reply:
we cannot really be sure of this here in cases where we also support 'reading' other tags into weldx properties (e.g. asdf/astropy quantities)

Right now I have implemented this manually in places where we support other tags
https://github.com/BAMWelDX/weldx/blob/96d9bbcc4065009c3414230932131deb935c63db/weldx/tags/units/pint_quantity.py#L40-L42

This code is probably a more reasonable default so there is one less thing to worry about in the Converter implementation (and *if* we ever want to write out a non weldx tag we can implement this in the converter)

To be extra save we could also guard the list sorting like so
```python
def select_tag(self, obj, tags, ctx): 
    weldx_tags = [tag for tag in tags if tag.startswith(WELDX_TAG_URI_BASE)]
    if weldx_tags:
        return sorted(weldx_tags)[-1]
    return sorted(tags)[-1] 
```